webook/calendar_buddy/factory_manager.py

```diff
@@ -0,0 +1,142 @@
+    """
+        Get the fabrication hooks associated with the given context type, these will be run after fabrication
+        in the order as retrieved 
+
+        :param context_type: The context type of which to get hooks for
+        :type context_type: CalendarContext
+
+        :return: Returns a list of callable hooks. Will be empty if no hooks are registered.
+        :rtype: list of functions
+     """
+    hooks = []
+    for hook in _FACTORY_HOOKS:
+        if hook["context_type"] == context_type:
+            hooks.append(hook["hook"])
+    return hooks
+
+
+def register_context_factory(factory: base.BaseCalendarContextFactory, context_type: CalendarContext) -> None:
+    """
+        Register a new context factory, under a given context_type. There can only be one factory per context type.
+
+        :param factory: The factory to register
+        :type factory: Must be a derivation of BaseCalendarContextFactory
+
+        :param context_type: The context type this factory applies to
+        :type context_type: CalendarContext
+    """
+    _CONTEXT_FACTORIES[context_type.value] = factory
+
+
+def fabricate_calendar_context(context_type: CalendarContext) -> object:
+    """
+        Fabricate a new calendar context for the specified context_type
+        Returns a calendar context of the type specified. E.g FullCalendarContext, if context_type is FULLCALENDAR.
+        All contexts follow base.BaseCalendarContext
+
+        :param context_type: Which context_type to fabricate a context for
+        :type context_type: CalendarContext
+    """
+
+    context_factory = _get_context_factory_for_context_type(context_type)
+
+    if context_factory is None:
+        raise FactoryNotFoundException(context_type=context_type)
+
+    constructed_context = context_factory.fabricate()
+
+    for hook in _get_hooks_for_context_type(context_type):
+        constructed_context = hook(constructed_context)
+
+    return constructed_context
+
+
+def register_fabrication_hook(hook: Callable, context_type: CalendarContext) -> None:
+    """
+        Register a fabrication hook, to be run after creating a new calendar context, on the fabricated context.
+        This allows you to append your own changes to the fabrication process of this context, in a global fashion.
+
+        :param hook: The function that is to be registered, and later run on fabrications. Will always receive the context, and should always return the context.
+        :type hook: Callable, function
+
+        :param context_type: Designates which context this fabrication hook applies to. The factory serving the designated context will run the hook you supplied.
+        :type context_type: CalendarContext 
+    """
+    _FACTORY_HOOKS.append({
+        "hook": hook,
+        "context_type": context_type
+    })
+
+
+def register_defaults (defaults: dict(), context_type: CalendarContext, standard_type: StandardType) -> None:
```

Comment:
dont use typehints as a dict literal, use Dict from typing or just `dict`
you would use it like so `dict[str, str]`